### PR TITLE
Jetty 9.4.x #1918 scalable scheduler

### PR DESCRIPTION
**Pull request by gregw**
A proposed solution for #1918 improving HttpClient scheduling.  This is a remake of PR #1934 with correct signoff.

This has a CyclicTimeoutTask class that implements the IdleTimeout mechanism in a style similar to a regular scheduler.

The HttpClient has been modified to use the CyclicTimeoutTask:

The HttpRequest class now takes a timestamp when the request is sent, so that all timeouts can be relative to this time.
The HttpDestination does a regular sweep (using a CyclicTimeoutTask) of all queued exchanges looking for expired timeouts
The HttpConnection class has its own CyclicTimeoutTask which is applied to the current exchange.

### CODE REVIEW

**Comment by gregw** [comment]
@sbordet Most of the tests are failing because sometimes the tests expect TimeoutException to be wrapped in an ExcecutionException, and others don't?    Since the method signature lists TimeoutException, I'm trying to be consistent and always throw that unwrapped... but I'm not sire if that is a change in the contract?

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpRequest.java [hunk 1]
```diff
@@ -677,14 +678,31 @@ public ContentResponse send() throws InterruptedException, TimeoutException, Exe
...
+            // TimeoutException to always throwing a ExecutionException(TimeoutException).
+            // Thus for backwards compatibility we unwrap the timeout exception here
+            if (x.getCause() instanceof TimeoutException)
+            {
+                TimeoutException t = (TimeoutException) (x.getCause());
+                abort(t);
+                throw t;
+            }
 
-            return listener.get(timeout, TimeUnit.MILLISECONDS);
```
Comment (gregw) [inline jetty-client/src/main/java/org/eclipse/jetty/client/HttpRequest.java]: @sbordet I think this was a race that was causing a lot of the HttpClient test instability.  By using the request timeout as the timeout on the `future.get`, we have two timeouts racing each other, but with different results.  One throws a TimeoutException and the other throws an ExecutionException wrapping a TimeoutException.     I don't think we need both, so I have removed the `future.get` timeout, as it is needless extra scheduling.   But I am also now unwrapping the exectionexception to try to keep the API behaviour the same... see comment in the new code.

**Comment by sbordet** [comment]
Gahh, I think I messed up because I rebased this branch locally and then pushed my additional changes.

**Comment by joakime** [comment]
All of those extra commits looks like a merge from jetty-9.4.x to <your_branch>

**Comment by joakime** [comment]
You cannot rebase / squash this branch now, as it contains commits from others (eg: zachary-duquette)

**Comment by joakime** [comment]
Try just closing this PR and opening a new one?

**Comment by gregw** [comment]
Won't the other commits just be merged out if this is squashed? They are already in the main branch with full credit so they won't be lost?

**Comment by sbordet** [comment]
@gregw yes, but the problem was that trying to review this PR showed all the other commits.
The last commit I made was forced after a rebase, so that it only shows the commits related to the work on this branch.

**Pull request closed without merging**